### PR TITLE
Fix issues on sharing a folder from the share extension

### PR DESCRIPTION
**Pull request by adrien-coye**
### Abstract

Sharing a folder from `Files.app` now creates a zip file and uploads it.

Linked [core PR](https://github.com/Infomaniak/ios-core/pull/46)

### CODE REVIEW

File: kDriveCore/Utils/AppGroupPathProvider.swift [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import Foundation
+import OSLog
+
+// TODO: Move to core /all
```
Comment (adrien-coye) [inline kDriveCore/Utils/AppGroupPathProvider.swift:22]: TODO move to core

File: kDriveCore/Utils/NSItemProvider+Detect.swift [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+ This program is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+// TODO: Move to Core /all
```
Comment (adrien-coye) [inline kDriveCore/Utils/NSItemProvider+Detect.swift:19]: TODO: Move to core

File: kDriveCore/Utils/UTI.swift [hunk 3]
```diff
@@ -16,6 +16,8 @@
  along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
+// TODO: Move to Core /all
```
Comment (adrien-coye) [inline kDriveCore/Utils/UTI.swift:19]: Move to core

File: kDriveCore/Utils/FileImportHelper.swift [hunk 4]
```diff
@@ -223,20 +240,76 @@ public class FileImportHelper {
...
+                            DDLogError("could not move zipped file:\(error)")
+                            progress.completedUnitCount += perItemUnitCount
+                            errorCount += 1
+                            dispatchGroup.leave()
+                            return
+                        }
+                    }
+                }
+
+            case .isMiscellaneous, .none:
```
Comment (adrien-coye) [inline kDriveCore/Utils/FileImportHelper.swift:312]: I have an issue with arbitrary types, that should be handeled as a `file`. Working on a fix.

Comment (adrien-coye) [reply kDriveCore/Utils/FileImportHelper.swift]: fix

**Review by PhilippeWeidmann — APPROVED** [review]

**Pull request merged**